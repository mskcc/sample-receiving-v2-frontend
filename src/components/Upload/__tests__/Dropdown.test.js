import React, { Component } from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import TestDropdown from '../formComponents/Dropdown'

const onChange = jest.fn()
const Dropdown = props => <TestDropdown onChange={onChange} {...props} />

describe('Render Dropdown', () => {
  it('render Dropdown correctly', () => {
    const DropdownComponent = renderer
      .create(<Dropdown/>)
      .toJSON()
    expect(DropdownComponent).toMatchSnapshot()
  })

  it('render Dropdown correctly with loading animation', () => {
    const wrapper = mount(<Dropdown loading={true} />)
    const progress = wrapper.find('LinearProgress').first()
    expect(progress.exists()).toBe(true)
  })
})
