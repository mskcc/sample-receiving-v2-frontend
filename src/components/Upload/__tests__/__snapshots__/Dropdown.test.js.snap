// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Dropdown render Dropdown correctly 1`] = `
<div
  className="Dropdown-textField-1"
>
  <div
    aria-expanded={false}
    aria-haspopup="listbox"
    aria-labelledby="downshift-0-label"
    aria-owns={null}
    role="combobox"
  >
    <div
      className="MuiFormControl-root MuiFormControl-fullWidth"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink={false}
        htmlFor="downshift-0-input"
        id="downshift-0-label"
      >
        Missing translationId: upload.form.test_me_label for language: \${ languageCode }
      </label>
      <div
        aria-activedescendant={null}
        aria-autocomplete="list"
        aria-controls={null}
        aria-labelledby="downshift-0-label"
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoComplete="off"
          className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
          disabled={false}
          id="test_me"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          required={false}
          type="text"
          value=""
        />
        <div
          className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
        >
          <button
            aria-label="Toggle menu open"
            className="MuiButtonBase-root MuiIconButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
      <p
        className="MuiFormHelperText-root"
      >
        Missing translationId: upload.form.test_me_helptext for language: \${ languageCode }
      </p>
    </div>
  </div>
</div>
`;
